adapters/alpies-eth/index.js

```diff
@@ -0,0 +1,115 @@
+require("dotenv").config();
+
+const moment = require("moment");
+const BigNumber = require("bignumber.js");
+const Ethereum = require("../../sdk/EVMC");
+const axios = require("axios");
+const URL = "http://nft-sales-service.dappradar.com/open-source";
+const KEY = process.env.DAPPRADAR_API_KEY;
+const path = require("path");
+
+class AlpiesETH {
+    constructor() {
+        this.name = "Alpies";
+        this.symbol = "ALPIES";
+        this.token = "0x0000000000000000000000000000000000000000";
+        this.protocol = "ethereum";
+        this.block = 13531146;
+        this.contract = "0x57a7c5d10c3f87f5617ac1c60da60082e44d539e";
+        this.events = ["Transfer"];
+        this.pathToAbi = path.join(__dirname, "./abi.json");
+        this.range = 500;
+        this.chunkSize = 6;
+        this.sdk = new Ethereum(this);
+    }
+
+    run = async () => {
+        const s = await this.getSymbol();
+        this.sdk = this.loadSdk();
+        this.symbol = s;
+        await this.sdk.run();
+    };
+
+    loadSdk = () => {
+        return new Ethereum(this);
+    };
+    getSymbol = async () => {
+        const resp = await axios.get(
+            `${URL}/token-metadata?key=${KEY}&token_address=${this.token}&protocol=${this.protocol}`,
+            {
+                headers: {
+                    "User-Agent":
+                        "Mozilla/5.0 (Macintosh; Intel Mac OS X 11_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4619.141 Safari/537.36",
+                },
+            },
+        );
+        const symbol = resp.data;
+        return symbol;
+    };
+    getPrice = async timestamp => {
+        const resp = await axios.get(
+            `${URL}/token-price?key=${KEY}&token_address=${this.token}&protocol=${this.protocol}&timestamp=${timestamp}`,
+            {
+                headers: {
+                    "User-Agent":
+                        "Mozilla/5.0 (Macintosh; Intel Mac OS X 11_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4619.141 Safari/537.36",
+                },
+            },
+        );
+        return resp.data;
+    };
+    stop = async () => {
+        this.sdk.stop();
+    };
+
+    getBuyer = async event => {
+        const buyer = event.returnValues.owner;
+        if (event.event === "Transfer") {
+            const txReceipt = await this.sdk.getTransactionReceipt(event.transactionHash);
+            if (txReceipt === null) {
+                return null;
+            }
+            return txReceipt.from;
+        }
+        return buyer;
+    };
+
+    process = async event => {
+        const block = await this.sdk.getBlock(event.blockNumber);
+        const timestamp = moment.unix(block.timestamp).utc();
+        const po = await this.getPrice(block.timestamp);
+        const nativePrice = new BigNumber(event.returnValues.cost).dividedBy(10 ** this.symbol.decimals);
```

Comment:
event.returnValues.cost is undefined and does not pass the tests.

Reply:
fixed. 
"Max key limit reached" is return when run test. Does this mean I need to apply for a new api key?